libpromises/evalfunction.c

```diff
@@ -8601,6 +8666,16 @@ static const FnCallArg READFILE_ARGS[] =
     {NULL, CF_DATA_TYPE_NONE, NULL}
 };
 
+static const FnCallArg CLASSEXPRESSION_FILTERDATA_ARGS[] =
+{
+    {CF_ABSPATHRANGE, CF_DATA_TYPE_STRING, "File name to read"},
+    {CF_VALRANGE, CF_DATA_TYPE_INT, "Column with class expression"},
+    {CF_ANYSTRING, CF_DATA_TYPE_STRING, "String to delimit data by"},
```

Comment:
Is this a literal string delimiter, or is it a regular expression?

Reply:
Currently this is not used. The problems I see with allowing you to choose the delimiters are:

1. To change this in `csv_parser.c`, the code has to be refactored a bunch. This will probably take a bit of time, since the code has a lot of spaghetti.
2. If it is to be done "on top" of the CSV parsing -- as in replacing occurrences of a given delimiter with `,` -- you can end up with situations where one line suddenly has more delimiters than intended because a comma was used as well.

I do however think that it should be changed. Allowing CFEngine to read CSV-files with non-comma delimiters would be a good idea. It's just that it will probably take more time, and if it should work for this function, it should also work for `readcsv()`.

Reply:
2 is no good, we would have to change the csv parser as you mention in 1. I think you should remove this function param, it can be added as a last optional parameter later.